charts/team-ns/templates/external-secrets.yaml

```diff
@@ -2,41 +2,52 @@
 {{- $ := . }}
 {{- if get (index $v.apps "external-secrets") "enabled" }}
 {{- range $s := $v.secrets }}
-{{- $ns := index $s "namespace" | default $.Release.Namespace }}
+{{- $clusterWide := $s.clusterWide | default false }}
+{{- $teamsWide := $s.teamsWide | default false }}
+{{- $hasNamespaces := (gt (len ($s.namespaces | default list)) 0) }}
+{{- $hasTeams := (gt (len ($s.teams | default list)) 0) }}
+{{- $isCluster := or $clusterWide $teamsWide $hasNamespaces $hasTeams }}
 {{- $pathPrefix := printf "secret/data/teams/team-%s" $v.teamId }}
 {{- $vaultPath := printf "%s/%s" $pathPrefix $s.name | quote }}
 {{- $teamNamespacesCollection := print "team-" (include "helm-toolkit.utils.joinListWithSep" (dict "list" $v.teamIds "sep" ",team-")) }}
+{{- $teamNamespacesSelection := print "team-" (include "helm-toolkit.utils.joinListWithSep" (dict "list" ($s.teams | default list) "sep" ",team-")) }}
+{{- $namespacesSelection := (include "helm-toolkit.utils.joinListWithSep" (dict "list" ($s.namespaces | default list) "sep" ",")) }}
 
 apiVersion: external-secrets.io/v1beta1
-{{- if and (eq $v.teamId "admin") (or ($s.clusterWide | default false) ($s.teamWide | default false)) }}
+{{- if $isCluster }}
 kind: ClusterExternalSecret
 {{- else }}
 kind: ExternalSecret
 {{- end }}
 metadata:
-  {{- if and (eq $v.teamId "admin") (or ($s.clusterWide | default false) ($s.teamWide | default false)) }}
-    {{- if $s.clusterWide | default false }}
+  {{- if $clusterWide }}
   name: otomi-cw-{{ $s.name }}
-    {{- else }}
+  {{- else if $teamsWide }}
   name: otomi-tw-{{ $s.name }}
-    {{- end }}
+  {{- else if $hasTeams }}
+  name: otomi-team-{{ $v.teamId }}-{{ $s.name }}
   {{- else }}
   name: {{ $s.name }}
   {{- end }}
-  namespace: {{ $ns }}
   labels: {{- include "team-ns.chart-labels" $ | nindent 4 }}
     {{- with $s.labels }}
     {{- . | toYaml | nindent 4 }}
     {{- end }}
 spec:
-  {{- if and (eq $v.teamId "admin") (or ($s.clusterWide | default false) ($s.teamWide | default false)) }}
+  {{- if $isCluster }}
   namespaceSelector:
-    {{- if $s.clusterWide | default false }}
+    {{- if $clusterWide }}
     matchLabels: 
       {}
-    {{- else }}
+    {{- else if $teamsWide }}
     matchExpressions:
       - {key: name, operator: In, values: [{{ $teamNamespacesCollection }}]}
+    {{- else if $hasNamespaces }}
+    matchExpressions:
+      - {key: name, operator: In, values: [{{ $namespacesSelection }}]}
```

Comment:
each team each team has a common namespace label: `type: team`

Reply:
of course...doh!